Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -9,44 +9,69 @@
         {
-            var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPath);
-            if (objects == null)
-                return;
+            internal const string kAssetPath = "ProjectSettings/InputManager.asset";
+            internal const string kAssetName = InputSystem.kProjectWideActionsAssetName;
+
+            // DONE 1. Implement reading the kAssetPath into InputActionAsset.
+            // DONE 2. Serialize as JSON and write as an .inputactions file into Asset directory.
+            // TODO Consider preserving GUIDs to potentially enable references to stay intact.
+            // TODO 3. Let InputActionImporter do its job on importing and configuring the asset.
+            // TODO 4. Assign to InputSystem.actions
 
-            // Handle deleting all InputActionAssets as older 1.8.0 pre release could create more than one project wide input asset in the file
-            foreach (var obj in objects)
+            class ProjectSettingsPostprocessor : AssetPostprocessor
             {
-                if (obj is InputActionReference)
+                static void OnPostprocessAllAssets(string[] importedAssets, string[] deletedAssets, string[] movedAssets, string[] movedFromAssetPaths, bool didDomainReload)
                 {
-                    var actionReference = obj as InputActionReference;
-                    AssetDatabase.RemoveObjectFromAsset(obj);
-                    Object.DestroyImmediate(actionReference);
+                    MoveInputManagerAssetActionsToProjectWideInputActionAsset();
                 }
-                else if (obj is InputActionAsset)
+            }
+
+            internal static void MoveInputManagerAssetActionsToProjectWideInputActionAsset()
+            {
+                var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPath);
+                if (objects != null)
                 {
-                    AssetDatabase.RemoveObjectFromAsset(obj);
+                    var inputActionsAsset = objects.FirstOrDefault(o => o != null && o.name == kAssetName) as InputActionAsset;
+                    if (inputActionsAsset != null)
+                    {
+                        var json = JsonUtility.ToJson(inputActionsAsset, prettyPrint: true);
+                        File.WriteAllText(ProjectWideActionsAsset.kDefaultAssetPath, json);
+                    }
                 }
+
+                DeleteActionAssetAndActionReferences();
             }
-        }
-    }
 
-    internal static class ProjectWideActionsAsset
-    {
-        private const string kDefaultAssetPath = "Assets/InputSystem_Actions.inputactions";
-        private const string kDefaultTemplateAssetPath = "Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsTemplate.json";
+            internal static void DeleteActionAssetAndActionReferences()
+            {
+                var objects = AssetDatabase.LoadAllAssetsAtPath(kAssetPath);
```

Comment:
This looks redundant, either pass `objects` array from `MoveInputManagerAssetActionsToProjectWideInputActionAsset()` into this function or combine the functions together since I believe its the only place where this is used?

Reply:
@AlexTyrer

Reply:
Good call - I've removed the DeleteActionAssetAndActionReferences() function and just put the implementation inside MoveInputManagerAssetActionsToProjectWideInputActionAsset().

Commit: aee68129f9765b77ca559bcc0c9c39aa3f7b1a72